int32 numMotors
MotorState[] states
